sycl/test/basic_tests/image_accessor_readwrite.cpp

```diff
@@ -1,8 +1,9 @@
 // RUN: %clangxx -fsycl %s -o %t.out
 // RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
-// RUNx: %GPU_RUN_PLACEHOLDER %t.out
-// RUNx: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// UNSUPPORTED: cuda
```

Comment:
I'm not sure I understand why this test is disabled on CUDA?
How "supporting more image_channel_types in image_accessors read/write API for half4 datatype on **host device**", might impact CUDA device?
Could you add a comment with clarification here, please?

Reply:
It seems like some functionality was enabled for OpenCL devices as well. Right?

Reply:
Short Answer: The test was edited to enable testing on GPUs, since CUDA does not support images currently, it was disabled for it.

Detailed Answer:
Two changes were done in this test: removal of commented half datatype testing (added separate test) and correcting the code to enable the test on GPUs (Commit#2).
The correction done is: removing the check on the exact written value; instead we read it and check the read value. The check is removed because OpenCL CPU and OpenCL GPU returns different value when the minimum value is expected to be written. 
Eg: In line 237, 238 of the unedited file image_accessor_readwrite.cpp CPU returns -128, but OpenCL GPU returns -127. Both the values are considered correct. I had created an defect on this for OpenCL GPU driver a long time ago, and we came to the consensus that the returned value of -127 is also correct and as per the spec. 
So, instead of making the value check more complicated I removed it and decided to just check the read value.
Checking the written value wasn't doing a more precise check; so removing it seemed fine.
It was added to aid debugging and enable a better understanding of how images are written and read for anyone interested.

Reply:
Please, either move commit#2 to separate PR or update this PR title and description to reflect the full scope of the changes.
I was confused because title says about changes for the host device and there is no description.

Reply:
description updated. Please approve and merge.